### PR TITLE
Add common functionality required for all in-tree to CSI Migration shim code pieces

### PR DESCRIPTION
**Pull request by davidz627**
Many of the CSI Migration shim pieces depend on some common functionality, mostly revolving around feature flags and this plugin interface method. Would be nice to get these pieces in first so that the work on different pieces of migration can continue in parallel without rebasing issues.

/kind feature
/sig storage
/assign @saad-ali @jsafrane 
/cc @ddebroy @leakingtapan @dims 

```release-note
NONE
```

### CODE REVIEW

**Comment by davidz627** [comment]
/cc @msau42 @kubernetes/sig-storage-pr-reviews 
xref: https://github.com/kubernetes/enhancements/issues/625

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 1]
```diff
@@ -75,11 +76,11 @@ func NewOperationGenerator(kubeClient clientset.Interface,
...
 	return &operationGenerator{
-		kubeClient:                       kubeClient,
-		volumePluginMgr:                  volumePluginMgr,
-		recorder:                         recorder,
+		kubeClient:      kubeClient,
+		volumePluginMgr: volumePluginMgr,
+		recorder:        recorder,
 		checkNodeCapabilitiesBeforeMount: checkNodeCapabilitiesBeforeMount,
-		blkUtil:                          blkUtil,
+		blkUtil: blkUtil,
```
[the remark this answers is not in the document]
Comment (davidz627) [reply pkg/volume/util/operationexecutor/operation_generator.go]: I think my gofmt must be wonky, I've been getting a lot of weird diffs like this that I have to fix manually

**Comment by davidz627** [comment]
@saad-ali fixed the gofmt issue, could you re-lgtm

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**